test/EndToEndSpec.hs

```diff
@@ -0,0 +1,38 @@
+module EndToEndSpec (spec) where
+
+import           Test.Hspec                                    (Spec, it, shouldBe)
+import           Data.Text                                     (Text)
+import qualified Data.Aeson                                    as JSON
+import qualified Data.Aeson.Types                              as JSON.T
+import qualified Language.PureScript.CoreFn.FromJSON           as CoreFn
+import           Language.PureScript.CoreFn.Module             (Module)
+import           Language.PureScript.CoreFn.Ann                (Ann)
+import           Language.PureScript.Scheme.CodeGen.Transpiler (moduleToScheme)
+import           Language.PureScript.Scheme.CodeGen.Printer    (printScheme)
+import           Language.PureScript.Scheme.CodeGen.Optimizer  (runOptimizations)
+
+readJSONFile :: FilePath -> IO (Module Ann)
+readJSONFile path = do
+  r <- JSON.decodeFileStrict' path
+  let res = r >>= JSON.T.parseMaybe CoreFn.moduleFromJSON
+  case res of
+    Nothing -> error "Nope"
+    Just (_version, module_) -> return module_
+
+compile :: FilePath -> IO Text
+compile path = (printScheme . runOptimizations . moduleToScheme)
+           <$> readJSONFile path
+
+spec :: Spec
+spec = do
```

Comment:
Instead of versioning the corefn files, we should just have a PureScript project (somewhere like `test/test-project`) and compile it before running the tests. Something like:
```suggestion
setup = do
  oldDir <- Turtle.pwd
  Turtle.cd "test/test-project"
  (exitCode, out, err) <- Turtle.shellStrictWithErr "spago build -u '-g corefn'"
  Turtle.cd oldDir
  case exitCode of
    ExitSuccess -> pure ()
    ExitFailure -> error "spago failed to run"

spec = beforeAll_ setup $ do
```

Then you could just read the resulting corefn files from `output`